frontend/app/src/store/trade/wallet.ts

```diff
@@ -0,0 +1,334 @@
+import type {
+  GasFeeEstimation,
+  PrepareERC20TransferResponse,
+  PrepareNativeTransferResponse,
+  RecentTransaction,
+  TransactionError,
+  TransactionParams,
+} from '@/types/trade';
+import { useTradeApi } from '@/composables/api/trade';
+import { useAssetInfoRetrieval } from '@/composables/assets/retrieval';
+import { useSupportedChains } from '@/composables/info/chains';
+import { useWalletHelper } from '@/composables/trade/wallet-helper';
+import { WagmiAdapter } from '@reown/appkit-adapter-wagmi';
+import { type AppKitNetwork, arbitrum, base, bsc, gnosis, mainnet, optimism, polygon, scroll } from '@reown/appkit/networks';
+import { type AppKit, createAppKit, useAppKitProvider } from '@reown/appkit/vue';
+import { assert, bigNumberify } from '@rotki/common';
+import { startPromise } from '@shared/utils';
+import { BrowserProvider, formatUnits, type TransactionResponse } from 'ethers';
+
+export const ROTKI_DAPP_METADATA = {
+  description: 'Rotki Dapp',
+  icons: ['https://raw.githubusercontent.com/rotki/data/refs/heads/main/assets/default_icons/website_logo.png'],
+  name: 'Rotki Dapp',
+  url: 'https://rotki.com',
+};
+
+export const EIP155 = 'eip155';
+
+export const supportedNetworks: [AppKitNetwork, ...AppKitNetwork[]] = [
+  mainnet,
+  base,
+  arbitrum,
+  optimism,
+  bsc,
+  gnosis,
+  polygon,
+  scroll,
+] as const;
+
+function buildAppKit(): AppKit {
+  const projectId = import.meta.env.VITE_WALLET_CONNECT_PROJECT_ID as string;
+
+  const wagmiAdapter = new WagmiAdapter({
```

Comment:
we don't use this in the website right? what is the difference of this and the ethers adapter? do we need both?

Reply:
Okay now when I tried again, I think it doesn't work with SPA, but works with nuxt.
I got error `process is not defined` smth.